runtime/foucoco/src/xcm_config.rs

```diff
@@ -64,29 +66,31 @@ pub struct CurrencyIdConvert;
 
 impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
 	fn convert(id: CurrencyId) -> Option<MultiLocation> {
-		match id {
-			CurrencyId::XCM(xcm_assets::RELAY) => Some(MultiLocation::parent()),
-			// Moonbase testnet native token
-			CurrencyId::XCM(xcm_assets::MOONBASE_DEV) => Some(moonbase_alpha::DEV_location()),
-			CurrencyId::Native => Some(native_location_external_pov()),
-			_ => None,
-		}
+		AssetRegistry::metadata(&id)
+			.filter(|m| m.location.is_some())
+			.and_then(|m| m.location)
+			.and_then(|l| l.try_into().ok())
 	}
 }
 
 impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
-	fn convert(location: MultiLocation) -> Option<CurrencyId> {
-		match location {
-			loc if loc == MultiLocation::parent() => Some(xcm_assets::RELAY_id()),
-			// Our native currency location without re-anchoring
-			loc if loc == native_location_external_pov() => Some(CurrencyId::Native),
-			// Our native currency location with re-anchoring
-			// The XCM pallet will try to re-anchor the location before it reaches here
-			loc if loc == native_location_local_pov() => Some(CurrencyId::Native),
-			// Moonbase testnet native token
-			loc if loc == moonbase_alpha::DEV_location() => Some(xcm_assets::MOONBASE_DEV_id()),
-			_ => None,
-		}
+	fn convert(location: MultiLocation) -> Option<CurrencyId>  {
+		let para_id = ParachainInfo::parachain_id();
+
+		let unanchored_location = match location {
+            MultiLocation { parents: 0, interior } => {
+     
+                match interior.pushed_front_with(Parachain(u32::from(para_id))) {
+                    Ok(new_interior) => MultiLocation {
+                        parents: 1,
+                        interior: new_interior,
+                    },
+                    Err(_) => return None, 
+                }
+            },
+            x => x,
+        };
+		AssetRegistry::location_to_asset_id(unanchored_location)
 	}
 }
```

Comment:
Can we try to generify this struct and put it into `runtime/common`? I think the only problem is the `ParachainInfo::parachain_id()` that is specific per runtime but we might be able to supply that to the struct instead and then we can have just one struct for all runtimes.

Reply:
Yes! We can leave `ParachainInfo` as generic. Good idea.

Reply:
I pushed a new [commit](https://github.com/pendulum-chain/pendulum/pull/432/commits/ddabc241039ce524d50be7912648fba229118a2a) for this. I had to leave `CurrencyId` defined in the common implementation due to some restrictions with the generics, but since all chains share it, I think it should be fine.